### ISSUE TITLE
Storage configuration

### ISSUE THREAD

**Issue by pat0s** [covertsh/ubuntu-autoinstall-generator#6]
Hi, 
it is possible to configure storage interactively during installation using your method?

**Comment by covertsh** [comment]
Hi, yes it is totally possible. Here is an example user-data file with interactive storage (modified from the user-data.example file in this repository):

```
#cloud-config
autoinstall:
  version: 1
  interactive-sections:
   - storage
  identity:
    hostname: ubuntu-server
    password: "$6$exDY1mhS4KUYCE/2$zmn9ToZwTKLhCw.b4/b.ZRTIZM30JZ4QrOQ2aOXJ8yk96xpcCof0kxKwuX1kqLG/ygbJ1f8wxED22bTL4F46P0"
    username: ubuntu
```

Have a look at https://ubuntu.com/server/docs/install/autoinstall-reference where you will find documentation for the user-data file, including how to make sections interactive during installation.

**Comment by pat0s** [comment]
Hi,
thank you for your response. Yes I also found it on ubuntu webpage. Everything worked just fine but when I tried to reboot this error occured. 
![error_kernel_isolinux_txt_cfg](https://user-images.githubusercontent.com/58843354/111912807-99b76980-8a6b-11eb-8aa4-c0e8fde29a12.PNG)

Have you tried your method with the latest version of ubuntu server, version 20.04.2?

**Comment by covertsh** [comment]
In my testing I am able to replicate this. It appears that one file is having its name truncated for some reason. Inside the orginal ISO file, there is a file "pool/main/l/linux-hwe-5.8/linux-modules-extra-5.8.0-48-generic_5.8.0-48.54~20.04.1_amd6.deb". In the generated ISO, it is named "linux-modules-extra-5.8.0-48-generic_5.8.0-48.54~20.04.1_amd6.de" (notice it is missing a "b" at the end). I have no idea why that would happen. At first I thought it could be a path length limit but I don't think the path is long enough to cause that. Still investigating.

**Comment by pat0s** [comment]
Ok. Thank you. I don't know what can cause it. Now I use a legacy version
in order to set storage configuration interactively during installation.
So after changing this file name everything works fine?

Dňa št, 25. mar 2021 18:34 covertsh ***@***.***> napísal(a):

**Comment by covertsh** [comment]
I think there is actually a problem with the ISO images being published by the Ubuntu project at the moment. The file is named ```linux-modules-extra-5.8.0-48-generic_5.8.0-48.54~20.04.1_amd6.deb``` in the downloaded ISO (verified with SHA256 hash and GPG signature) but the ```md5sum.txt``` file contains the line ```d09244a003ee46b5d2e81160f044c465  ./pool/main/l/linux-hwe-5.8/linux-modules-extra-5.8.0-48-generic_5.8.0-48.54~20.04.1_amd64.deb```. (see amd**6** instead of amd**64**) This means that whoever packaged up the ISO image at Ubuntu made a small mistake in the name of the file, which causes a warning at the start of the installation.

There is a second problem which is the truncation of the final "b" at the end of the filename when using my script. I have made a commit to enable long file names in the mkisofs command options which has fixed that.

Lastly, I see the following error in the subiquity installer log after the crash when I reproduce the issue:

![image](https://user-images.githubusercontent.com/45894800/112537173-4b290880-8da6-11eb-9d1d-5efb92491696.png)

This does not happen in a fully-automated installation, perhaps because the reporting is disabled? (I believe that is what controls the HTTP requests). Unfortunately according to the autoinstall reference, the ```reporting``` setting is ignored if any sections are interactive. I think this is why you are getting a crash and sadly I don't think we can fix that without the Ubuntu making a fix for the bug in subiquity. I might be wrong about why this is crashing but that is what my troubleshooting suggests for now. The only thing I suggest is to not use interactive sections for the time being, but that might not be suitable for you. Instead you could use an older ISO if you have one already and tell the script to use that with the ```-s``` option.

**Comment by pat0s** [comment]
Thank you for your help. I really appreciate it. I can't use fully automated installation. But I am still able to use legacy vesion of ubuntu server 20.04.1. It uses old preseed method. As long as this error exist, it is the best choice for me.

**Comment by covertsh** [comment]
I have tested this again today with the latest ISO and the issue is still present unfortunately.

**Comment by covertsh** [comment]
Hooray! I tested again today with the latest daily ISO (20.04.3) and the user-data file I posted as an example for interactive storage configuration worked. At the end, the reboot worked fine and did not crash the installer. I think that issue has been fixed.

However, the MD5 check still gives an error because the ```linux-modules-extra``` file is still being renamed with missing characters. This does not cause an issue during the installation and the checks can also be disabled with the ```-c``` option for the generator script. I won't close this issue for now while I continue to investigate that, but the interactive storage problem seems to be resolved.

**Comment by skupfer** [comment]
@covertsh I ran into the same issue with a file missing "d64", namely `/pool/main/l/linux-hwe-5.11/linux-modules-extra-5.11.0-27-generic_5.11.0-27.29~20.04.1_am.deb`. I am using WSL2 and mounted the original 20.04.3 server iso (which I feed to the script) where the file is present with the full name. When running the script I checked the /tmp/ folder and noticed the file got the wrong name in there, too. So this is caused by a faulty extraction of the image itself.

Tried it in a native ubuntu 20 vm and the same happens there, so it is not wsl2-related. The filename gets shortened and therefore wrongly repacked.
